scipy/sparse/linalg/eigen/arpack/arpack.py

```diff
@@ -1864,13 +1865,22 @@ def matmat_XH_X(x):
 
         eigvals, eigvec = lobpcg(XH_X, X, tol=tol ** 2, maxiter=maxiter,
                                  largest=largest)
-
+    elif solver == 'propack':
+        from scipy.sparse.linalg import svdp
+        jobu = ((isinstance(return_singular_vectors, str) and 'u' in return_singular_vectors)
+                or (isinstance(return_singular_vectors, bool) and return_singular_vectors))
+        jobv = ((isinstance(return_singular_vectors, str) and 'v' in return_singular_vectors)
+                or (isinstance(return_singular_vectors, bool) and return_singular_vectors))
+        return svdp(
```

Comment:
@mdhaber Now that I look at this again, I'm not sure this is the best way to do this.  Past me probably did this to avoid recalculating the singular values, but it also means we don't take advantage of any of the checking/fixing that happens below.  There is some checking happening in `svdp` function itself, but might be worth reviewing best way to integrate this as a solver

Reply:
I think you've done a reasonable thing. The notes of `svds` say:
>    This is a naive implementation using ARPACK or LOBPCG as an eigensolver
    on A.H * A or A * A.H, depending on which one is more efficient.

This needs to be updated, of course, but it says that all of this below is needed because ARPACK and LOBPCG are not designed specifically for SVD; they're for eigenvalue problems. PROPACK is for SVD, so it makes sense to me to return its SVD directly rather than mess with it.